features/customers.feature

```diff
@@ -3,8 +3,27 @@ Feature: The customer store service back-end
     I need a RESTful customer service
     So that I can keep track of all my customers
 Background:
-    Given the following customers
-        | username | password | first_name | last_name | addresses |
-        | user1    | 1234     | Henry      | George    | []        |
-        | user2    | 1235     | John       | Georgie   | []        |
-        | user3    | 1236     | Tom        | Giovani   | []        |
\ No newline at end of file
+    Given the following customers 
+        | username | password | first_name | last_name | street_address| city        | state       | zipcode | country |
+        | user1   | 1234     | Henry      | George    | 123           | New York    | NY          | 10000   | USA     |
+        | user2   | 1235     | John       | Georgie   | 234           | Boston      | Massachusat | 20000   | USA     |
+        | user3   | 1236     | Tom        | Giovani   | 345           | White plain | NY          | 10567   | USA     |
+
+Scenario: The server is running
+    When I visit the "Home Page"
+    Then I should see "Customers RESTful Service" in the title
+    And I should not see "404 Not Found"
+
+Scenario: Create a customer
+    When I visit the "Home Page"
+    And I set the "username" to "dome"
+    And I set the "password" to "1111"
+    And I set the "first_name" to "jiraphon"
+    And I set the "last_name" to "yenphraphai"
+    And I set the "street_address" to "1234"
+    And I set the "city" to "New York"
+    And I set the "state" to "NY"
+    And I set the "zip" to "10090"
+    And I set the "country" to "USA"
+    And I press the "Create Customer" button
+    Then I should see the message "Success"
```

Comment:
I'm wondering if we should add in tests for the error conditions ("username" is taken, missing one customer attribute, and missing just one address attribute)

Reply:
Sure